dockerfiles/docs.Dockerfile

```diff
@@ -16,8 +16,9 @@ COPY --from=hugo $GOPATH/bin/hugo /bin/hugo
 WORKDIR /src
 
 FROM build-base AS build
+ARG DOCS_BASEURL
 RUN --mount=type=bind,rw,source=docs,target=. \
-    hugo --gc --minify --destination /out
+    hugo --gc --minify --destination /out -b $DOCS_BASEURL
```

Comment:
From the other PR; https://github.com/distribution/distribution/pull/4104#discussion_r1358506534

> what happens when building without passing a base-URL? will `-b` without value cause it to fail?
> 
> (what's the default if the option is not set? or should the ARG have a default (eg `localhost`) ?

If we need a default, it may be better to set `ARG DOCS_BASEURL=/` (and remove the default from the `docker-bake.hcl` ?) /cc @crazy-max

Reply:
That's exactly what we now do, and for the build we pull the right base path from the github pages url and don't include the hostname at all.

Reply:
Oh wait, move the default, could do that :)

I really know nothing about docker bake.

Reply:
Yeah, so if _possible_ I always try to make Dockerfiles themselves to have a sensible default, so that they can "work" without passing a `--build-arg`. That way a `docker build .` wouldn't fail (but of course defaults can be overridden in CI, or other ways)

Reply:
Agree with @thaJeztah, building without bake would lead to erroneous hugo command because `DOCS_BASEURL` would be empty.

Better to set a bake variable block in the bake definition and defaults build arg to `ARG DOCS_BASEURL=/`:

Reply:
I'm sure @milosgajdos would appreciate a push to the branch.

Reply:
Oh, you wrote it up in a way even I can understand, I'll test+push ;)

Reply:
I don't have perms to push to his branch but here is the commit https://github.com/crazy-max/docker-distribution/commit/9f73f976f5464b4dbd6d510facb8e1f495617e89

Reply:
```
$ docker buildx bake docs-export --print
{
  "group": {
    "default": {
      "targets": [
        "docs-export"
      ]
    }
  },
  "target": {
    "docs-export": {
      "context": ".",
      "dockerfile": "./dockerfiles/docs.Dockerfile",
      "target": "out",
      "output": [
        "type=local,dest=build/docs"
      ]
    }
  }
}
```

```
$ DOCS_BASEURL=/foo docker buildx bake docs-export --print
{
  "group": {
    "default": {
      "targets": [
        "docs-export"
      ]
    }
  },
  "target": {
    "docs-export": {
      "context": ".",
      "dockerfile": "./dockerfiles/docs.Dockerfile",
      "args": {
        "DOCS_BASEURL": "/foo"
      },
      "target": "out",
      "output": [
        "type=local,dest=build/docs"
      ]
    }
  }
}
```

Reply:
Can you see what I've done wrong here?
https://github.com/Jamstah/distribution/blob/26bfe0fef852d09d68822e9edf96d4b5f8a992ef/.github/workflows/docs.yml
https://github.com/Jamstah/distribution/actions/runs/6518929917/job/17704881302

It didn't pick up the env var and used the default: http://jammy.id/distribution/

Reply:
See my commit https://github.com/distribution/distribution/pull/4102#discussion_r1359492783

Reply:
I'm AFK but @Jamstah please do push to this branch as you see fit.

Reply:
Ah yes, missed your comment first time, thanks. Changes pushed.